plugins/woocommerce/includes/admin/notes/class-wc-notes-run-db-update.php

```diff
@@ -19,20 +19,43 @@ class WC_Notes_Run_Db_Update {
 	const NOTE_NAME = 'wc-update-db-reminder';
 
 	/**
-	 * Attach hooks.
+	 * Checks whether the note needs an update based on the current db update status.
+	 * Hooked onto the 'woocommerce_get_note_from_db' filter. See {@see \WC_Install}.
+	 *
+	 * @since 10.3.0
+	 *
+	 * @param Note $note Note to check.
+	 * @return Note
 	 */
-	public function __construct() {
-		// If the old notice gets dismissed, also hide this new one.
-		add_action( 'woocommerce_hide_update_notice', array( __CLASS__, 'set_notice_actioned' ) );
-
-		// Not using Jetpack\Constants here as it can run before 'plugin_loaded' is done.
-		if ( defined( 'DOING_AJAX' ) && DOING_AJAX
-			|| defined( 'DOING_CRON' ) && DOING_CRON
-			|| ! is_admin() ) {
-			return;
+	public static function maybe_update_notice( $note ) {
+		if ( ! $note instanceof Note || $note->get_name() !== self::NOTE_NAME ) {
+			return $note;
+		}
+
+		if ( ! in_array( 'update', \WC_Admin_Notices::get_notices(), true ) ) {
```

Comment:
I'm not one for a lot of inline comments, but this seems like a good case to either add them or piece these out into methods that describe what's happening.  It definitely takes a minute to parse what's happening in each of these conditions.